core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java

```diff
@@ -125,6 +126,17 @@ private UnderFileSystem getOrAdd(AlluxioURI ufsUri, UnderFileSystemConfiguration
         return cachedFs;
       }
       UnderFileSystem fs = UnderFileSystem.Factory.create(ufsUri.toString(), ufsConf);
+
+      // Detect whether to use managed blocking on UFS operations.
+      boolean useManagedBlocking = fs.isObjectStorage();
```

Comment:
There could be a worse design with a better Ceph performance like below:
boolean useManagedBlocking = (fs.getUnderFSType() != "s3");

Reply:
If the Ceph performance is not substantially worse, I think we can keep it as is

Reply:
Ceph just lost the extra gain from ForkJoinPool for add/delete. List is still better than before. So overall it's either on par or better.